packages/webcrypto-ed25519-polyfill/src/secrets.ts

```diff
@@ -131,10 +131,54 @@ export function verifyPolyfill(key: CryptoKey, signature: BufferSource, data: Bu
     if (key.type !== 'public' || !key.usages.includes('verify')) {
         throw new DOMException('Unable to use this key to verify', 'InvalidAccessError');
     }
-    const publicKeyBytes = ed25519.getPublicKey(getSecretKeyBytes_INTERNAL_ONLY_DO_NOT_EXPORT(key));
+    const storedBytes = getSecretKeyBytes_INTERNAL_ONLY_DO_NOT_EXPORT(key);
     try {
-        return ed25519.verify(bufferSourceToUint8Array(signature), bufferSourceToUint8Array(data), publicKeyBytes);
+        return (
+            ed25519.verify(
+                bufferSourceToUint8Array(signature),
+                bufferSourceToUint8Array(data),
+                ed25519.getPublicKey(storedBytes)
+            ) || ed25519.verify(bufferSourceToUint8Array(signature), bufferSourceToUint8Array(data), storedBytes)
+        );
     } catch {
         return false;
     }
 }
+
+export function importKeyPolyfill(
+    format: KeyFormat,
+    keyData: BufferSource,
+    extractable: boolean,
+    keyUsages: readonly KeyUsage[]
+): CryptoKey {
+    const type =
+        // could be more rigorous
+        format === 'pkcs8' ? 'private' : 'public';
+
+    const bytes = bufferSourceToUint8Array(keyData);
+
+    const keyBytes =
+        format === 'pkcs8'
+            ? // Remove header
+              bytes.slice(16)
+            : bytes;
+
+    if (type === 'public') {
+        const publicKey = {
+            [Symbol.toStringTag]: 'CryptoKey',
+            algorithm: Object.freeze({ name: 'Ed25519' }),
+            extractable,
```

Comment:
I think that like in `generatePolyfill` a public key should always be extractable. Probably just ignore the user's input here to match the behaviour there